openlibrary/catalog/marc/marc_binary.py

```diff
@@ -130,7 +130,7 @@ def read_isbn(self, f):
         if '\x1f' in f.line:
             return super(MarcBinary, self).read_isbn(f)
         else:
-            m = re_isbn.match(line[3:-1])
```

Comment:
There is a way that _undefined names_ can work if _script_a.py_ declares a variable in a global space and then later on in the same interpreter run _script_b.py_ uses that same variable.  That is of course fragile and implicit and not recommended.  However _undefined names_ are a sign of dead code that never/rarely gets executed.

Reply:
Do we leave it like this or remove the code?

Reply:
@YBthebest I've looked at this code trying to figure out what it is trying to do, and why it has not caused a problem before.   I don't understand what situation matching `line[3:-1]` is meant to cover. The conditional is testing "if the current MARC isbn field has a subfield" ... use the parent `MarcBase.read_isbn()`

_All_ MARC ISBN 020 fields should have sub-fields... so my best guess is that this is meant to try and extract and isbn looking value from a corrupt record, but I'm not sure it is very robust, and clearly it has been broken. I think removing the entire `read_isbn()` and simply using the parent `MarcBase.read_isbn()` would be the fix.

This class and code _is_ used though, so it might be wise to do it in a separate PR and maybe add some more tests?

Actually, that's my advice. Leave it as is (your correction matches the intent of the original), but raise an issue to follow up and remove a hopefully unnecessary attempt at extract isbn if the data is bad. I'd rather flag corrupt data than attempt to run forward with it as this is a common source of bad text and bad isbns persisting in our records.

Reply:
I have raised #1681 for this